### PR TITLE
[browserstack-service] Sending skipped test details for beforeEach, beforeAll and afterEach hooks in mocha

### PR DESCRIPTION
**Pull request by sriteja777**
## Proposed changes
 - Sending test details as skipped whenever hook fails along with hook failure details

Related v7 PR https://github.com/webdriverio/webdriverio/pull/10259

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-browserstack-service/src/util.ts [hunk 1]
```diff
@@ -329,7 +329,14 @@ export function getUniqueIdentifier(test: Frameworks.Test, framework?: string):
     if (framework === 'jasmine') {
         return test.fullName
     }
-    return `${test.parent} - ${test.title}`
+
+    let parentTitle = test.parent
+    // Sometimes parent will be an object instead of a string
+    if (typeof test.parent === 'object') {
+        // @ts-ignore
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/util.ts:336]: Can we resolve the type issue?

**Review by christian-bromann — COMMENTED** [review]
Can we update the branch and see if the test pass? If so I am happy to merge.

File: packages/wdio-browserstack-service/src/util.ts (hunk 1 above)
Comment (sriteja777) [reply packages/wdio-browserstack-service/src/util.ts]: Actually, we are [getting the test object](https://github.com/webdriverio/webdriverio/blob/1ce750d11ad8e14486ce92a77935ebb442d11956/packages/wdio-browserstack-service/src/insights-handler.ts#L127) from the context of the test that is passed to the service (`test.ctx.test.parent`) in some places. And this object doesn't match with `Framework.Test` type which has parent has string, but in the context object, the parent can be an object. Also we don't have any exported types to type for this context.

Currently removed the ts-ignore and casted it to `any`. Let me know if you have any better solution

**Comment by sriteja777** [comment]
@christian-bromann the tests have passed. Please merge if it looks good to you.

**Review by christian-bromann — APPROVED** [review]
LGTM 👍

**Pull request merged**